packages/frontend/editor-ui/src/features/dataStore/components/dataGrid/ColumnHeader.vue

```diff
@@ -118,15 +157,24 @@ const onHeaderClick = (event: MouseEvent) => {
 			</div>
 		</div>
 
+		<N8nIconButton
+			v-show="areButtonsVisible"
+			icon="funnel"
+			type="tertiary"
+			text
+			:class="{ 'filter-highlighted': hasActiveFilter }"
+			@click="onShowFilter"
+		/>
+
 		<N8nActionDropdown
-			v-show="isDropdownVisible"
-			ref="dropdownRef"
+			v-show="areButtonsVisible"
 			data-test-id="data-store-column-header-actions"
 			:items="columnActionItems"
 			:placement="'bottom-start'"
 			:activator-icon="'ellipsis'"
 			@select="onItemClick"
 			@visible-change="onDropdownVisibleChange"
+			@click="onDropdownClick"
```

Comment:
If the sole reason for `@click` handler is to stop event from propagating, this can just be `@click.prevent`